algebra/elliptic-curve/src/curve.rs

```diff
@@ -25,16 +25,39 @@ impl Affine {
     }
 
     #[must_use]
-    // TODO: return result
-    pub fn x(&self) -> FieldElement {
+    pub fn x(&self) -> Option<&FieldElement> {
         match self {
-            Self::Zero => panic!("no x coordinate for 0"),
-            Self::Point { x, .. } => x.clone(),
+            Self::Zero => None,
+            Self::Point { x, .. } => Some(x),
         }
     }
 
     #[must_use]
-    pub fn on_curve(&self) -> bool {
+    pub fn y(&self) -> Option<&FieldElement> {
+        match self {
+            Self::Zero => None,
+            Self::Point { y, .. } => Some(y),
+        }
+    }
+
+    #[must_use]
+    pub fn as_coordinates(&self) -> Option<(&FieldElement, &FieldElement)> {
```

Comment:
Not sure if there's a convention for this, but i think this should just be `coordinates`?